### PR TITLE
Update makefile to download arm64 version of vale

### LINKED ISSUE #3070: Update makefile to support downloading arm64 version of vale
**Bug description**

It is not possible to run the task `make lint` on an `arm64` architecture machine. When creating the release for the arm64 version of vale, the maintainers changed the naming convention. To download the release, we need to tweak our makefile to account for this inconsistency.

amd64 - `vale_2.20.1_macOS_64-bit.tar.gz`
arm64 - `vale_2.20.1_macOS_arm64.tar.gz`

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- (X) Test and Release
- ( ) User Experience
- (X) Developer Experience

**Expected behavior**

The make task should run.

**Steps to reproduce the bug**

run `make lint` on an Apple Silicon Mac

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by seemiller**
### What this PR does / why we need it

The command to download the `vale` application that is used in the `lint` make task needs to support `arm64`.

### Which issue(s) this PR fixes

Fixes #3070 

### Describe testing done for PR

[vuil] on amd and apple silicon macs:
```
git clean -dfx .
make tools
```

### Release note

```release-note

```

### CODE REVIEW

File: hack/tools/Makefile [hunk 1]
```diff
@@ -114,7 +114,7 @@ vale: $(VALE)
 $(VALE):
 	mkdir -p $(BIN_DIR)
 	# vale uses macOS for darwin, Linux for linux and 64 for amd64, so manually map them
-	curl -sfL https://github.com/errata-ai/vale/releases/download/v$(VALE_VERSION)/vale_$(VALE_VERSION)_$(shell echo $(HOST_OS) | sed 's,darwin,macOS,g'| sed 's,linux,Linux,g')_$(shell echo $(HOST_ARCH) | sed 's,amd,,g')-bit.tar.gz | tar -xz -C /tmp/
+	curl -sfL https://github.com/errata-ai/vale/releases/download/v$(VALE_VERSION)/vale_$(VALE_VERSION)_$(shell echo $(HOST_OS) | sed 's,darwin,macOS,g'| sed 's,linux,Linux,g')_$(shell echo $(HOST_ARCH) | sed 's,amd,,g')$([ $(HOST_ARCH) = "arm64" ] && echo "" || echo "-bit").tar.gz | tar -xz -C /tmp/
```
Comment (marckhouzam) [inline hack/tools/Makefile:117]: I believe the new logic `$([ $(HOST_ARCH) = "arm64" ] && echo "" || echo "-bit")` needs to use the `shell` command first, or else the `-bit` does not appear when I tried it for `amd64`:

`$(shell [ $(HOST_ARCH) = "arm64" ] && echo "" || echo "-bit")`

**Review by marckhouzam — COMMENTED** [review]
Thanks for this @seemiller!
I was just trying to make the linter work on `darwin-arm64` without much luck.  This is a step in the right direction.

File: hack/tools/Makefile (hunk 1 above)
Comment (seemiller) [reply hack/tools/Makefile]: I added the missing `shell` to the command.

**Review by marckhouzam — APPROVED** [review]
LGTM

**Review by rajathagasthya — APPROVED** [review]

**Comment by marckhouzam** [comment]
Hiya @seemiller.  This bit me again today.  Any time to get this merged?

**Comment by vuil** [comment]
Thanks for the changes @seemiller. I hit the same issue a few days ago, was working on a fix when I remembered this. 
I took the liberty to rebase it, fixed up the comments and simplified the name substitution slightly.

**Pull request merged**